src/test/java/teammates/test/cases/browsertests/FeedbackMcqQuestionUiTest.java

```diff
@@ -291,4 +294,294 @@ public void testDeleteQuestionAction() {
+
+        // Uncheck checkboxes to maintain consistency among other tests
+        feedbackEditPage.clickMcqAssignWeightCheckboxForNewQuestion();
+        feedbackEditPage.clickAddMcqOtherOptionCheckboxForNewQuestion();
+
+        feedbackEditPage.clickDiscardChangesLinkForNewQuestion();
+        feedbackEditPage.waitForConfirmationModalAndClickOk();
+    }
+
+    @Test
+    public void testMcqWeightsFeature_shouldAddWeightsCorrectly() throws Exception {
+        ______TS("Success: Add weights");
+        feedbackEditPage.clickAddQuestionButton();
+        feedbackEditPage.selectNewQuestionTypeAndWaitForNewQuestionPanelReady("MCQ");
+        feedbackEditPage.fillQuestionTextBox("MCQ weight feature", NEW_QUESTION_INDEX);
+        feedbackEditPage.fillQuestionDescription("More details", NEW_QUESTION_INDEX);
+        feedbackEditPage.clickMcqAssignWeightCheckboxForNewQuestion();
+        feedbackEditPage.clickAddMcqOtherOptionCheckboxForNewQuestion();
+
+        // Fill MCQ choices and check corresponding weight values
+        feedbackEditPage.fillMcqOptionForNewQuestion(0, "Choice 1");
+        feedbackEditPage.fillMcqWeightBox(NEW_QUESTION_INDEX, 0, "1");
+        feedbackEditPage.fillMcqOptionForNewQuestion(1, "Choice 2");
+        feedbackEditPage.fillMcqWeightBox(NEW_QUESTION_INDEX, 1, "2");
+        feedbackEditPage.fillMcqOtherWeightBox(NEW_QUESTION_INDEX, "3");
+
+        feedbackEditPage.clickAddQuestionButton();
+        feedbackEditPage.waitForTextsForAllStatusMessagesToUserEquals(Const.StatusMessages.FEEDBACK_QUESTION_ADDED);
+        assertNotNull(BackDoor.getFeedbackQuestion(courseId, feedbackSessionName, 1));
+
+        ______TS("MCQ: Add more weights");
+        feedbackEditPage.clickEditQuestionButton(1);
+        feedbackEditPage.clickAddMoreMcqOptionLink(1);
+        assertTrue(feedbackEditPage.isElementPresent("mcqOption-2-1"));
+        assertTrue(feedbackEditPage.isElementPresent("mcqWeight-2-1"));
+        feedbackEditPage.fillMcqOption(1, 2, "Choice 3");
+        feedbackEditPage.fillMcqWeightBox(1, 2, "4");
+        feedbackEditPage.clickSaveExistingQuestionButton(1);
+        feedbackEditPage.waitForTextsForAllStatusMessagesToUserEquals(Const.StatusMessages.FEEDBACK_QUESTION_EDITED);
+
+        ______TS("MCQ: Failed to add weight due to invalid choice value");
+        feedbackEditPage.clickEditQuestionButton(1);
+        feedbackEditPage.clickAddMoreMcqOptionLink(1);
+
+        // Test for choice that has only spaces
+        assertTrue(feedbackEditPage.isElementPresent("mcqOption-3-1"));
+        assertTrue(feedbackEditPage.isElementPresent("mcqWeight-3-1"));
+        feedbackEditPage.fillMcqOption(1, 3, "                           "); // Choices with only spaces
+        feedbackEditPage.fillMcqWeightBox(1, 3, "5");
+
+        feedbackEditPage.clickSaveExistingQuestionButton(1);
+        feedbackEditPage.waitForTextsForAllStatusMessagesToUserEquals(Const.StatusMessages.FEEDBACK_QUESTION_EDITED);
```

Comment:
No error message?

Reply:
No. This is the normal behaviour for cases where there are invalid (space-only) choices. The question form just simply removes the choice cells with the invalid value. SO, this is also the default behaviour for the corresponding weight cell as well.

This is the reason, I am testing that the choice cell and the corresponding weight cell are not present in the question form after editing the question.

```java
        // Check that the two added choices and the corresponding weights are not present
        assertFalse(feedbackEditPage.isElementPresent("mcqOption-3-1"));
        assertFalse(feedbackEditPage.isElementPresent("mcqWeight-3-1"));
```

Reply:
Add comment to note it down